packages/web-components/@sfx/product/src/variant.ts

```diff
@@ -0,0 +1,46 @@
+import { customElement, property, html } from 'lit-element';
+import { ProductVariantModel } from './product';
+
+import { Base } from '@sfx/base';
+
+@customElement('sfx-product-variant')
+export default class Variant extends Base {
+  @property({ type: String }) type: string = 'text';
+  @property({ type: Object }) variant: ProductVariantModel = { text: '', product: {} };
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    const { type, variant } = this;
+
+    // Configure variant component
+    this.setAttribute('role', 'listitem');
+    this.classList.add('product-variant');
+
+    switch (type) {
+      case 'color':
+        this.setAttribute('style', `background-color:${variant.color}`);
+        this.setAttribute('title', variant.text);
+        break;
+      case 'image':
+        this.setAttribute('style', `background-image:${variant.image};background-color:${variant.color}`);
```

Comment:
Should the image be a background image? The image is meaningful content if it's an image variant, so I think it should be put in an `<img>` tag.